tests/unit-tests/test_dataloading.py

```diff
@@ -77,10 +83,474 @@
                                             prepare_batch_edge_input)
 from graphstorm.dataloading.utils import modify_fanout_for_target_etype
 from graphstorm.dataloading.utils import trim_data
+from graphstorm.dataloading.metadata import (GSGraphMetadata,
+                                             GSMetadataDglDistGraph,
+                                             config_json_sanity_check,
+                                             load_metadata_from_json)
 
-from numpy.testing import assert_equal, assert_almost_equal
-from transformers import AutoTokenizer
-from torch.utils.data import DataLoader
+def build_gcons_json_example():
```

Comment:
Can we have a separate test file, e.g., test_metagraph.py ?

Reply:
Separate new tests into a `test_metadata.py`.